File: engine/src/main/java/me/centauri07/menu/engine/core/Menu.java
package me.centauri07.menu.engine.core;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class Menu extends Option {
    private final Option[] options;

    public Menu(@NotNull Component component, @NotNull Option[] options) {
        super(component);

        Objects.requireNonNull(options);

        this.options = options;
    }

    public Option[] getOptions() {
        return options;
    }
}

File: engine/src/main/kotlin/me/centauri07/menu/engine/Menu.kt
package me.centauri07.menu.engine

import me.centauri07.menu.engine.core.Component
import me.centauri07.menu.engine.core.Menu

fun menu(component: Component, block: MenuScope.() -> Unit): Menu {
    val scope = MenuScope(component)
    scope.block()

    return scope.construct()
}